msticpy/context/azure/sentinel_core.py

```diff
@@ -35,6 +39,59 @@
 
 logger = logging.getLogger(__name__)
 
+_SUB_ID = "subscription_id"
+_RES_GRP = "resource_group"
+_WS_NAME = "workspace_name"
+_RES_ID = "resource_id"
+
+
+@dataclass
+class SentinelInstanceDetails:
+    """Dataclass for Sentinel workspace details."""
+
+    subscription_id: Optional[str]
+    resource_group: Optional[str]
+    workspace_name: Optional[str]
+
+    @property
+    def resource_id(self) -> Optional[str]:
+        """Return the resource ID for the workspace."""
+        if not all([self.subscription_id, self.resource_group, self.workspace_name]):
+            return None
+        return build_sentinel_resource_id(
+            self.subscription_id, self.resource_group, self.workspace_name  # type: ignore
+        )
+
+
+def _get_ws_defaults(
+    *,
+    subscription_id: Optional[str] = None,
```

Comment:
Again, I don't think these should be Optional?

When this is called, all 3 arguments _must_ be present. Each time we partial this function, we can potentially add one or more of them, but that doesn't make them optional.

Reply:
Amendment: they might be Optional here if we also checked for "none of the args are passed, so I will do the default workspace lookup instead" before passing it into constructor for the SentinelInstanceDetails

Reply:
I've changed the logic to do this in _set_ws_defaults